### PR TITLE
Update Bigquery streaming sample.

### PR DESCRIPTION
**Pull request by jsimonweb**

### CODE REVIEW

File: bigquery/api/GettingStarted/BigqueryTest/BigqueryTest.cs [hunk 1]
```diff
@@ -48,6 +49,15 @@ public class BigQueryTest
             }
         };
 
+        internal class CustomTransientErrorDetectionStrategy
+            : ITransientErrorDetectionStrategy
+        {
+            public bool IsTransient(Exception ex)
```
Comment (SurferJeffAtGoogle) [inline bigquery/api/GettingStarted/BigqueryTest/BigqueryTest.cs]:
```cs
 public bool IsTransient(Exception ex) => ex is InvalidOperationException;
```

File: bigquery/api/GettingStarted/BigqueryTest/BigqueryTest.cs [hunk 2]
```diff
@@ -533,18 +537,36 @@ public void TestImportDataFromFile()
         public void TestImportDataFromStream()
         {
             string datasetId = "datasetForTestImportDataFromStream";
-            string newTableId = "tableForTestImportDataFromStream";
+            Random rnd = new Random();
+            string tableNameSuffix = rnd.Next(4200000).ToString();
```
Comment (SurferJeffAtGoogle) [inline bigquery/api/GettingStarted/BigqueryTest/BigqueryTest.cs]: Why pass 4200000 here?

File: bigquery/api/GettingStarted/BigqueryTest/BigqueryTest.cs [hunk 3]
```diff
@@ -533,18 +537,36 @@ public void TestImportDataFromFile()
...
+                        newTableId, query, _client);
+                    var row = results.First();
+                    valueToTest = row["title"].ToString();
+                });
+            }
+            catch (Exception)
+            {
+                // All of the retries failed.
+            }
+            Assert.Equal(gcsUploadTestWord, valueToTest);
```
Comment (SurferJeffAtGoogle) [inline bigquery/api/GettingStarted/BigqueryTest/BigqueryTest.cs]: If this assert fails, then DeleteTable and DeleteDataset won't be called below.  The test will leak resources.

**Comment by jsimonweb** [comment]
Addressed comments. Please review.

**Pull request merged**